examples/src/keyring/aws_kms/act_like_aws_kms_master_key_provider.py

```diff
@@ -0,0 +1,88 @@
+# Copyright Amazon.com Inc. or its affiliates. All Rights Reserved.
+# SPDX-License-Identifier: Apache-2.0
+"""
+Before there were keyrings, there were master key providers.
+Master key providers were the original configuration structure
+that we provided for defining how you want to protect your data keys.
+
+The AWS KMS master key provider was the tool that we provided for interacting with AWS KMS.
+Like the AWS KMS keyring,
+the AWS KMS master key provider encrypts with all CMKs that you identify,
+but unlike the AWS KMS keyring,
+the AWS KMS master key provider always attempts to decrypt
+*any* data keys that were encrypted under an AWS KMS CMK.
```

Comment:
```suggestion
the AWS KMS master key provider can decrypt
*any* data key that was encrypted under an AWS KMS CMK.
```
-or- 
```
the AWS KMS master key provider can use any AWS KMS CMK that encrypted
a data key to decrypt it. 
```

Reply:
That's not quite what I was getting at. Whether or not it *can* decrypt depends on the KMS/IAM permissions. Just like the discovery keyring, the point I was trying to make is that it does in fact reach out to the KMS service for every KMS-encrypted EDK whether or not it succeeds.

Reply:
The difficulty here is the combination of "always" and "any." Does the ESDK always attempt to decrypt all of the encrypted data keys in an encrypted message? Or, it can attempt to decrypt any of them?